### DISCUSSION TITLE
variable worker-cpu relationships (for targets)

### DISCUSSION THREAD

**Discussion by colebrookson** [wlandau/crew.cluster#34] [Q&A] [answered]
Hi, I'm trying to implement `crew.cluster` with `targets` to deploy a pipeline on a SLURM-based HPC cluster. I have been able to run targets on the HPC before with `clustermq` but am trying to transition to `crew.cluster`. My question is specifically about the interaction between the two - I'm curious how to deploy a targets pipeline which has variable requirements of CPUs per target. 

I have many targets that should just run on one CPU, but a few targets that are fitting models that use MCMC with multiple chains that I'd like to be run on multiple CPUs. How can I make it clear that targets X, Y and Z each require 8 CPU, whereas all other targets require only one CPU. Is this possible?

**Comment by wlandau** [comment, marked as answer]
In `crew`, you can define a controller group with different controllers which have different worker resources. Please see https://books.ropensci.org/targets/crew.html#heterogeneous-workers and https://wlandau.github.io/crew/articles/groups.html.
